packages/flutter/test/widgets/navigator_test.dart

```diff
@@ -627,6 +628,39 @@ void main() {
     expect(observations[2].previous, '/A');
   });
 
+  testWidgetsWithLeakTracking('$Route  dispatches memory events', (WidgetTester tester) async {
+    Future<void> createAndDisposeRoute() async {
+      final GlobalKey<NavigatorState> nav = GlobalKey<NavigatorState>();
+      await tester.pumpWidget(
+      MaterialApp(
+        navigatorKey: nav,
+        home: const Scaffold(
+          body: Text('home'),
+        )
+      )
+    );
+      const TestPage page = TestPage(name: 'page');
+      final Route<dynamic> route = page.createRoute(nav.currentContext!);
+      // ignore: invalid_use_of_protected_member
+      route.dispose();
```

Comment:
you can do
```dart
nav.currentState!.push(MaterialPageRoute(builder: (_) => const PlaceHolder())); // This should create a route
await tester.pumpAndSettle();

nav.currentState!.pop();
await tester.pumpAndSettle(); // this should dispose the route.

```

Reply:
nav.currentState is null on first line
Did i miss something?

Reply:
you will need this first before push
```dart
await tester.pumpWidget(
      MaterialApp(
        navigatorKey: nav,
        home: const Scaffold(
          body: Text('home'),
        )
      )
    );
```

Reply:
Applied. Thank you!